packages/adapter/src/lib/adapter/adapter.js

```diff
@@ -8983,6 +8978,34 @@ function Adapter(options) {
 
                 this.adapterConfig = adapterConfig;
 
+                this.log = new Log(this.namespaceLog, config.log.level, logger);
+
+                //
+                // From here on "this" methods can be used that might log with "this.log" !!
+                // Above this line only use logger!
+                //
+
+                await createInstancesObjects(adapterConfig);
+
+                // auto oObjects
+                if (options.objects) {
+                    this.oObjects = await this.getAdapterObjectsAsync();
+                    await this.subscribeObjectsAsync('*');
+                }
+
+                // read the systemSecret
+                if (systemSecret === null) {
+                    try {
+                        const data = await this.getForeignObjectAsync('system.config', null);
```

Comment:
I would also use adapterObjects here

Reply:
missed the obvious one :-)

Reply:
ääähhmm ... no this is initialized above or? ;-) ad also in the this.getEncryptedConfig is stuff from this ... so, I would leave that way

Reply:
yes it is, but why read with adapter internal methods to get the systemSecret? if we decide at some point to change the behavior of the adapter methods to protect some sensitive data which may be on the system config in the future then this would not be in there.. so checks in the adapter internal methods are just redundant.

Reply:
imho the methods are mainly for external use by adapter devs

Reply:
agreed